src/ray/gcs/tables.h

```diff
@@ -451,6 +451,129 @@ class Set : private Log<ID, Data>,
   using Log<ID, Data>::num_lookups_;
 };
 
+template <typename ID, typename Data>
+class HashInterface {
+ public:
+  using DataT = typename Data::NativeTableType;
+  using DataMap = std::unordered_map<std::string, std::shared_ptr<DataT>>;
+  using HashCallback =
+      std::function<void(AsyncGcsClient *client, const ID &id, const DataMap &pairs)>;
+  using HashRemoveCallback = std::function<void(AsyncGcsClient *client, const ID &id,
+                                                const std::vector<std::string> &keys)>;
+  using HashNotificationCallback = std::function<void(
+      AsyncGcsClient *client, const ID &id,
+      const GcsTableNotificationMode notification_mode, const DataMap &data)>;
+  using SubscriptionCallback = typename Log<ID, Data>::SubscriptionCallback;
```

Comment:
Document the args of these callback functions?

Reply:
I moved the document from `class Hash` to `class HashInterface` and also document the args for the callbacks.